### PR TITLE
include assert in validator test in pytest

### PR DESCRIPTION
**Pull request by MrMrRobat**
## Change Summary
rename try_assert.py -> test_assert_in_validators.py
add """PYTEST_DONT_REWRITE"""

## Checklist

* [x] Unit tests for the changes exist
* [ ] Tests pass on CI and coverage remains at 100%
* [x] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)

### CODE REVIEW

**Comment by samuelcolvin** [comment]
looking good, I'll push to fix it and make a few other tweaks.

**Comment by MrMrRobat** [comment]
ah, didn't see your comment and already pushed fix with some tweaks too

**Comment by samuelcolvin** [comment]
my mistake, your fix was better.

File: tests/test_assert_in_validators.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        def check_a(cls, v):
+            assert v == 'a', 'invalid a'
+            return v
+
+    assert Model(a='a').a == 'a'
+
+    with pytest.raises(ValidationError) as exc_info:
+        Model(a='snap')
+
+    assert exc_info.value.errors() == [{'loc': ('a',), 'msg': 'invalid a', 'type': 'assertion_error'}]
```
Comment (MrMrRobat) [inline tests/test_assert_in_validators.py:23]: but this will not display diff on failure :(

Comment (samuelcolvin) [reply tests/test_assert_in_validators.py]: yes, my mistake, I've undone the change.

**Comment by samuelcolvin** [comment]
thank you, sorry to mess with your PR 🙏

**Pull request merged**